File: go/general/1089. 复写零.go
package main

// 利用append的特性
func duplicateZeros(arr []int) {
	for i := 0; i < len(arr)-1; i++ {
		if arr[i] == 0 {
			arr = append(arr[:i+1], arr[i:len(arr)-1]...) // 精髓
			i++
		}
	}
}

File: python/general/1089. 复写零.py
from typing import List

class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        if arr.count(0) == 0:
            return arr
        res = []
        for i in arr:
            if i != 0:
                res.append(i)
            else:
                res.append(0)
                if len(arr) != len(res):
                    res.append(0)
                else:
                    break
            if len(arr) == len(res):
                break

        i = 0
        while i< len(arr):
            arr[i] = res[i]
            i+=1

